frontend/ui/screensaver.lua

```diff
@@ -170,17 +170,25 @@ function Screensaver:expandSpecial(message, fallback)
     if ui and ui.document then
         -- If we have a ReaderUI instance, use it.
         local doc = ui.document
-        currentpage = ui.view.state.page or currentpage
-        totalpages = doc:getPageCount() or totalpages
+        if (doc:hasHiddenFlows()) then
```

Comment:
Seems fine at a glance; that being said we don't have a method to abstract this away in Document @poire-z ?
```suggestion
        if doc:hasHiddenFlows() then
```

Reply:
Well, there is some calculations that depends on self.ui.toc (and some UI expected values), so I don't think we really need to put more of that in self.document (and get headaches making decisions :)